### DISCUSSION TITLE
How to include BearSSL

### DISCUSSION THREAD

**Discussion by LordOfDeadbush** [earlephilhower/arduino-pico#2470] [Q&A]
Hello,
I am trying to follow the signing section of the [OTA](https://arduino-pico.readthedocs.io/en/latest/ota.html) chapter in the documentation, and I can't seem to figure out how to include BearSSL. I was just wondering what the `#include` was? It doesn't seem to mention this on the docs.

Thank you so much,
Corn

**Comment by maxgerhardt** [comment]
The BearSSL specific C++ classes are defined in `BearSSLHelpers.h`. But if you just include `ArduinoOTA.h`, they will be auto-imported, too. A minimal sketch that compiles for a "Pico W" is

```cpp
#include <Arduino.h>
#include <ArduinoOTA.h>

BearSSL::PublicKey signPubKey(
  "-----BEGIN PUBLIC KEY-----\n"
  "MFswDQYJKoZIhvcNAQEBBQADSgAwRwJAXWRPQyGlEY+SXz8Uslhe+MLjTgWd8lf/\n"
  "nA0hgCm9JFKC1tq1S73cQ9naClNXsMqY7pwPt1bSY8jYRqHHbdoUvwIDAQAB\n"
  "-----END PUBLIC KEY-----\n"
);
BearSSL::HashSHA256 hash;
BearSSL::SigningVerifier sign( &signPubKey );

void setup() {
  Update.installSignature( &hash, &sign );
}

void loop() {
}
```

Obviously not functional until all the rest is filled in, but it compiles.
